components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -944,38 +938,15 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             };
         }
 
-        if (!!workspace.softDeleted) {
```

Comment:
workspaceServer.getWorkspace does not check softDeleted, so probably it should stay?

Reply:
added softDeleted check in the workspace service